Performance patches prerequisite

Spottedleaf opened the pull request:
Original patches:
https://github.com/Spottedleaf/Concrete/blob/bleeding/patches/server/0004-Util-patch.patch
https://github.com/Spottedleaf/Concrete/blob/bleeding/patches/server/0022-Improve-autosave-performance.patch
https://github.com/Spottedleaf/Concrete/blob/bleeding/patches/server/0023-Avoid-immediately-autosaving-newly-generated-loaded-.patch

See patch notes for details
The patches aren't always 1:1 ports, they may have been changed to improve behavior or performance in some areas

They've all been dumped in this PR since they tend to have dependencies on each other (most notably the mc-util patch)

electronicboy reviewed (commented):
- [x] Spottedleaf/Concrete:patches/server/0004-Util-patch.patch@bleeding 

- [ ] Spottedleaf/Concrete:patches/server/0022-Improve-autosave-performance.patch@bleeding 
- [x] Spottedleaf/Concrete:patches/server/0023-Avoid-immediately-autosaving-newly-generated-loaded-.patch@bleeding 
- [ ] Spottedleaf/Concrete:patches/server/0013-Per-player-view-distance-implementation.patch@bleeding 
- [ ] Spottedleaf/Concrete:patches/server/0015-Optimize-isOutsideRange-to-use-a-square-radius-check.patch@bleeding 
- [x] Spigot-Server-Patches/0420-Optimise-getChunkAt-calls-for-loaded-chunks.patch

electronicboy commented on scripts/importmcdev.sh:
```diff
@@ -106,6 +106,7 @@ done
 
 # dont forget \ at end of each line but last
 importLibrary com.mojang authlib com/mojang/authlib yggdrasil/YggdrasilGameProfileRepository.java
+importLibrary com.mojang datafixerupper com/mojang/datafixers/util Either.java
```
Either?!

Spottedleaf committed:
async fixup

Spottedleaf committed:
more fixups

Spottedleaf committed:
more fixup

Spottedleaf committed:
Performance patch prereq

Prereq changes for the coming storm of performance patches.
Includes optimising incremental saving

electronicboy reviewed (changes requested):
https://www.youtube.com/watch?v=woySeSNBL3o
